### PR TITLE
created models and routes

### PR DESCRIPTION
**Pull request by Horlakz**
created models for the discussion, replies and votes
then added the routes endpoint for the forum

note: I'm not sure this code is correct sha, I'll test it after I finish writing the contollers but before then these are the updated codes

### CODE REVIEW

**Comment by Horlakz** [comment]
sorry, I forgot to include that I have configured the database already

**Comment by struckchure** [comment]
@Horlakz This commit message is okay, for your own sake, try to keep the commits professional, it'll help

File: models/discussionModel.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    title: {
+      type: String,
+      required: [true, "Please Add a Discussion title"],
+    },
+    body: {
+      type: String,
+      required: [true, "Please Add a body text"],
+    },
+    attachments: {
```
Comment (struckchure) [inline models/discussionModel.js:23]: the attachments is a list of files

File: models/replyModel.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    discussion_id: {
+      type: Schema.Types.ObjectId,
+      required: false,
+      ref: "Discussion",
+    },
+    reply_id: {
+      type: String,
+      required: false,
+    },
+    reply: {
```
Comment (struckchure) [inline models/replyModel.js:23]: this is not supposed to be here

File: models/discussionModel.js (hunk 1 above)
Comment (struckchure) [inline models/discussionModel.js:23]: this should be able to take a list of files, like an array of files

File: models/voteModel.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    user_id: {
+      type: String,
+      required: true,
+    },
+    content_id: {
+      type: String,
+      required: true,
+    },
+    action: {
```
Comment (struckchure) [inline models/voteModel.js:18]: you should be able to use enums here, like having just 2 available options, any other thing would be considered as invalid

**Review by struckchure — APPROVED** [review]

File: models/voteModel.js (hunk 3 above)
Comment (Horlakz) [reply models/voteModel.js:18]: I checked oo, I don't know how to

File: models/discussionModel.js (hunk 1 above)
Comment (Horlakz) [reply models/discussionModel.js:23]: I totally forgot, I wanted to change it to array before I push, will fix it now

File: models/voteModel.js (hunk 3 above)
Comment (struckchure) [reply models/voteModel.js:18]: make it something like an options field, with only 2 choices

File: models/replyModel.js (hunk 2 above)
Comment (Horlakz) [reply models/replyModel.js:23]: yes, I just figured while following the uml diagram that there's no actual place to add text to replies
we need to redo this meeting, I don't really understand that part sha
but should I remove it?

File: models/voteModel.js (hunk 3 above)
Comment (Horlakz) [reply models/voteModel.js:18]: I don't know if a boolean can work?

Comment (Horlakz) [reply models/voteModel.js:18]: > make it something like an options field, with only 2 choices

alright, let me work on this, I think I know what to do now

File: models/replyModel.js (hunk 2 above)
Comment (Horlakz) [reply models/replyModel.js:23]: I'll remove it

**Pull request merged**